.ci/CI/src/Bootloader.jl

```diff
@@ -0,0 +1,380 @@
+    target_branch::String,
+    git_ci_tools_url::String="https://github.com/QEDjl-project/QuantumElectrodynamics.jl.git",
+    git_ci_tools_branch::String="dev",
+)
+    # verification script that no custom URLs are used in unit tests
+    if target_branch != "main"
+        push!(job_dict["stages"], "verify-unit-test-deps")
+        job_dict["verify-unit-test-deps"] = Dict(
+            "image" => "julia:1.10",
+            "stage" => "verify-unit-test-deps",
+            "script" => [
+                "apt update && apt install -y git",
+                "git clone --depth 1 -b $(git_ci_tools_branch) $(git_ci_tools_url) /tools",
+                "julia /tools/.ci/verify_env.jl",
+            ],
+            "interruptible" => true,
+            "tags" => ["cpuonly"],
+        )
+    end
+end
+
+function get_normal_unit_test(
+    version::String,
+    target_branch::String,
+    dev_package_name::AbstractString,
+    dev_package_version::AbstractString,
+    dev_package_path::AbstractString,
+    git_ci_tools_url::String,
+    git_ci_tools_branch::String,
+)::Dict
+    job_yaml = Dict()
+    job_yaml["stage"] = "unit-test"
+    job_yaml["variables"] = Dict(
+        "CI_DEV_PKG_NAME" => dev_package_name,
+        "CI_DEV_PKG_VERSION" => dev_package_version,
+        "CI_DEV_PKG_PATH" => dev_package_path,
+    )
+    job_yaml["image"] = "julia:$(version)"
+
+    script = [
+        "apt update && apt install -y git",
+        "git clone --depth 1 -b $(git_ci_tools_branch) $(git_ci_tools_url) /tmp/integration_test_tools/",
+    ]
+
+    if target_branch == "main"
+        push!(
+            script,
+            "julia --project=. /tmp/integration_test_tools/.ci/CI/src/SetupDevEnv.jl \${CI_PROJECT_DIR}/Project.toml NO_MESSAGE",
+        )
+    else
+        push!(
+            script,
+            "julia --project=. /tmp/integration_test_tools/.ci/CI/src/SetupDevEnv.jl \${CI_PROJECT_DIR}/Project.toml",
+        )
+    end
+
+    script = vcat(
+        script,
+        [
+            "julia --project=. -e 'import Pkg; Pkg.instantiate()'",
+            "julia --project=. -e 'import Pkg; Pkg.test(; coverage = true)'",
+        ],
+    )
+    job_yaml["script"] = script
+
+    job_yaml["interruptible"] = true
+    job_yaml["tags"] = ["cpuonly"]
+
+    return job_yaml
+end
+
+function get_nighlty_unit_test(
+    target_branch::String,
+    dev_package_name::AbstractString,
+    dev_package_version::AbstractString,
+    dev_package_path::AbstractString,
+    git_ci_tools_url::String,
+    git_ci_tools_branch::String,
+)
+    job_yaml = get_normal_unit_test(
+        "1",
+        target_branch,
+        dev_package_name,
+        dev_package_version,
+        dev_package_path,
+        git_ci_tools_url,
+        git_ci_tools_branch,
+    )
+    job_yaml["image"] = "debian:bookworm-slim"
+
+    if !haskey(job_yaml, "variables")
+        job_yaml["variables"] = Dict()
+    end
+    job_yaml["variables"]["JULIA_DONWLOAD"] = "/julia/download"
```

Comment:
Typo, appears in a bunch of places.
```suggestion
    job_yaml["variables"]["JULIA_DOWNLOAD"] = "/julia/download"
```